modules/util/portal-tools-db-support/src/main/java/com/liferay/portal/tools/db/support/commands/CleanServiceBuilderCommand.java

```diff
@@ -155,8 +157,15 @@ private void _deleteServiceComponentRows(
 	private void _dropTable(Connection connection, String tableName)
 		throws SQLException {
 
-		try (Statement statement = connection.createStatement()) {
-			statement.executeUpdate("DROP TABLE IF EXISTS " + tableName);
+		DatabaseMetaData databaseMetaData = connection.getMetaData();
+
+		try (Statement statement = connection.createStatement();
+			ResultSet rs = databaseMetaData.getTables(
+				null, null, tableName.toUpperCase(), new String[] {"TABLE"})) {
```

Comment:
Why there is a toUpperCase() call on the query? But no in the drop statement?
I think we should not call the toUpperCase(). Some dbms supports case sensitive table names, if we unconditionally toUpperCase it, we might miss the table.

Reply:
Hi @shuyangzhou,

I think you're right. Unfortunately, if I don't do toUpperCase(), it causes the tests to fail, because we use h2 database when running the tests. h2 database requires you to put the table name in uppercase (see https://www.h2database.com/javadoc/org/h2/engine/DbSettings.html#DATABASE_TO_UPPER). How should we go about resolving this issue without breaking the tests?

Reply:
I am not aware we have a h2 test env. Do you mean support internal test env?

Reply:
It's in the test files in the portal-tools-db-support module. See https://github.com/liferay/liferay-portal/blob/master/modules/util/portal-tools-db-support/src/test/java/com/liferay/portal/tools/db/support/util/DBTestUtil.java#L73-L74.

Reply:
The worst case is, we have to run the query twice. First run with the exact given name, if we get it, just continue with it. If we don't get it, try upper case name as a fallback.
But really I don't even know h2 is on my support matrix. Please double check on that. And check whoever added that comment line, to see why h2 is under consideration at all.